Library/ViewModels/ProjectPamphletViewModelTests.swift

```diff
@@ -373,4 +375,32 @@ final class ProjectPamphletViewModelTests: TestCase {
     self.goToRewardsProject.assertValues([project], "Tapping 'Back this project' emits the project")
     self.goToRewardsRefTag.assertValues([.discovery], "Tapping 'Back this project' emits the refTag")
   }
+
+  func testConfigurePledgeCTAView_featureEnabled() {
+    let config = Config.template |> \.features .~ [Feature.checkout.rawValue: true]
+    let project = Project.template
+
+    withEnvironment(config: config) {
+      self.vm.inputs.configureWith(projectOrParam: .left(project), refTag: .discovery)
+      self.vm.inputs.viewDidLoad()
+      self.vm.inputs.viewWillAppear(animated: false)
```

Comment:
Commenting out both `viewWillAppear` + `viewDidAppear` in both of these tests passes...do we still need it then?

Reply:
I think we usually consider it good practice to trigger all the lifecycle inputs so it reproduces the sequence that would usually happen in a real app run.

Reply:
I see.

I think there's another HWDWWD candidate as I'm pretty sure this has been followed very loosely. I think keeping those calls here makes the test harder to read and actually factually incorrect (because it isn't dependent on those signals) but it does resemble the lifecycle closely so I'll stay undecided for now 😅

Reply:
Just chatted with @justinswart and wonder if we can assert that `configurePledgeCTAView` hasn't emitted before calling any inputs? To make sure that it indeed is one of the inputs causing the signal to emit.